src/hotspot/share/opto/superword.cpp

```diff
@@ -2359,472 +2052,6 @@ void SuperWord::schedule_reorder_memops(Node_List &memops_schedule) {
-        }
-      }
-#endif
-
-      phase()->register_new_node_with_ctrl_of(vn, first);
-      for (uint j = 0; j < p->size(); j++) {
-        Node* pm = p->at(j);
-        igvn().replace_node(pm, vn);
-      }
-      igvn()._worklist.push(vn);
-
-      if (vlen > max_vlen) {
-        max_vlen = vlen;
-      }
-      if (vlen_in_bytes > max_vlen_in_bytes) {
-        max_vlen_in_bytes = vlen_in_bytes;
-      }
-      VectorNode::trace_new_vector(vn, "SuperWord");
-    }
-  }//for (int i = 0; i < body().length(); i++)
-
-  if (max_vlen_in_bytes > C->max_vector_size()) {
-    C->set_max_vector_size(max_vlen_in_bytes);
-  }
-  if (max_vlen_in_bytes > 0) {
-    cl->mark_loop_vectorized();
-  }
-
-  if (SuperWordLoopUnrollAnalysis) {
-    if (cl->has_passed_slp()) {
-      uint slp_max_unroll_factor = cl->slp_max_unroll();
-      if (slp_max_unroll_factor == max_vlen) {
-#ifndef PRODUCT
-        if (TraceSuperWordLoopUnrollAnalysis) {
-          tty->print_cr("vector loop(unroll=%d, len=%d)\n", max_vlen, max_vlen_in_bytes*BitsPerByte);
-        }
-#endif
-        // For atomic unrolled loops which are vector mapped, instigate more unrolling
-        cl->set_notpassed_slp();
-        // if vector resources are limited, do not allow additional unrolling
-        if (Matcher::float_pressure_limit() > 8) {
-          C->set_major_progress();
-          cl->mark_do_unroll_only();
-        }
-      }
-    }
-  }
-
-  phase()->C->print_method(PHASE_SUPERWORD3_AFTER_OUTPUT, 4, cl);
-
-  return true;
-}
-
-//------------------------------vector_opd---------------------------
-// Create a vector operand for the nodes in pack p for operand: in(opd_idx)
-Node* SuperWord::vector_opd(Node_List* p, int opd_idx) {
-  Node* p0 = p->at(0);
-  uint vlen = p->size();
-  Node* opd = p0->in(opd_idx);
-  CountedLoopNode *cl = lpt()->_head->as_CountedLoop();
-  bool have_same_inputs = same_inputs(p, opd_idx);
-
-  // Insert index population operation to create a vector of increasing
-  // indices starting from the iv value. In some special unrolled loops
-  // (see JDK-8286125), we need scalar replications of the iv value if
-  // all inputs are the same iv, so we do a same inputs check here.
-  if (opd == iv() && !have_same_inputs) {
-    BasicType p0_bt = velt_basic_type(p0);
-    BasicType iv_bt = is_subword_type(p0_bt) ? p0_bt : T_INT;
-    assert(VectorNode::is_populate_index_supported(iv_bt), "Should support");
-    const TypeVect* vt = TypeVect::make(iv_bt, vlen);
-    Node* vn = new PopulateIndexNode(iv(), igvn().intcon(1), vt);
-    VectorNode::trace_new_vector(vn, "SuperWord");
-    phase()->register_new_node_with_ctrl_of(vn, opd);
-    return vn;
-  }
-
-  if (have_same_inputs) {
-    if (opd->is_Vector() || opd->is_LoadVector()) {
-      if (opd_idx == 2 && VectorNode::is_shift(p0)) {
-        assert(false, "shift's count can't be vector");
-        return nullptr;
-      }
-      return opd; // input is matching vector
-    }
```

Comment:
Note: since here the inputs were already transformed, we got input `VectorNode` or `LoadVectorNode`, for all members of the pack this same vector node. But now that we model this with `VTransform`, the input is also a pack. We handle this in `SuperWordVTransformBuilder::find_input_for_vector` with:
`Node_List* pack_in = _packset.isa_pack_input_or_null(pack, j);`